test/rbac_deploy.js

```diff
@@ -0,0 +1,198 @@
+        alice,
+        (result) => { addr = result.contract.address; },
+    );
+
+    const contract = new ContractPromise(api, contractData, addr);
+
+    await contractTransaction(
+        contract.tx.addUserToGroup(
+            { value: 0, gasLimit: 100000 * 1000000 },
+            EMPLOYEES.Tanisha, GROUPS.PeaqOffice,
+        ),
+        alice,
+    );
+
+    await contractTransaction(
+        contract.tx.addUserToGroup(
+            { value: 0, gasLimit: 100000 * 1000000 },
+            EMPLOYEES.Anton, GROUPS.PeaqOffice,
+        ),
+        alice,
+    );
+
+    // Check group
+    let callValue = [];
+    callValue = await contract.query.readUserGroup(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        GROUPS.PeaqOffice,
+    );
+    assert.equal(callValue.output.length, 2);
+    callValue.output.forEach((element) => {
+        assert.notEqual([EMPLOYEES.Tanisha, EMPLOYEES.Anton].indexOf(element.toHex()), -1);
+    });
+
+    // UserGroup assign role
+    await contractTransaction(
+        contract.tx.addUserOrGroupToRole(
+            { value: 0, gasLimit: 100000 * 1000000 },
+            GROUPS.PeaqOffice, ROLES.AccessToOffice,
+        ),
+        alice,
+    );
+
+    await contractTransaction(
+        contract.tx.addUserOrGroupToRole(
+            { value: 0, gasLimit: 100000 * 1000000 },
+            EMPLOYEES.Leo, ROLES.AccessToOffice,
+        ),
+        alice,
+    );
+
+    [GROUPS.PeaqOffice, EMPLOYEES.Leo].forEach(async (userGroup) => {
+        callValue = await contract.query.readUserOrGroupRoles(
+            alice.address,
+            { value: 0, gasLimit: -1 },
+            userGroup,
+        );
+        assert.equal(callValue.output.length, 1);
+        assert.equal(callValue.output[0].toHex(), ROLES.AccessToOffice);
+    });
+
+    // Assign Perm
+    await contractTransaction(
+        contract.tx.addRoleToPermission(
+            { value: 0, gasLimit: 100000 * 1000000 },
+            ROLES.AccessToOffice, PERMS.GrantMainDoorUnlock,
+        ),
+        alice,
+    );
+
+    callValue = await contract.query.readPermissions(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        ROLES.AccessToOffice,
+    );
+    assert.equal(callValue.output.length, 1);
+    assert.equal(callValue.output[0].toHex(), PERMS.GrantMainDoorUnlock);
+
+    // Checking
+    [EMPLOYEES.Tanisha, EMPLOYEES.Anton, EMPLOYEES.Leo].forEach(async (employee) => {
+        callValue = await contract.query.checkAccess(
+            alice.address,
+            { value: 0, gasLimit: -1 },
+            employee,
+            PERMS.GrantMainDoorUnlock,
+        );
+        assert.equal(callValue.output.valueOf(), true);
+    });
+
+    callValue = await contract.query.checkAccess(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        EMPLOYEES.Maryna,
+        PERMS.GrantMainDoorUnlock,
+    );
+    assert.equal(callValue.output.valueOf(), false);
+
```

Comment:
the api connection should be closed here else it will be left hanging and affect the node server
`api.disconnect();`